sap/5.3/modules/ROOT/pages/sap-connector-examples.adoc

```diff
@@ -1,104 +1,188 @@
 = SAP Connector 5.3 Examples - Mule 4
 :page-aliases: connectors::sap/sap-connector-examples.adoc
 
-== Establish an SAP Connection
+== Send an IDoc to SAP
+
+This example shows you how to create a Mule app that sends an IDoc to SAP. The following screenshot shows the Studio flow for this example:
+
+image::sap-connector-example-send-idoc.png[Studio flow for sending an IDoc to SAP]
 
-Create a Mule app that establishes a connection to SAP and receives a function using the SAP connector's *Function source* operation, which returns the export parameter to SAP.
+=== Configure HTTP Listener
+
+Configure HTTP Listener to initiate a Mule flow when a call is made to the `/sendIDoc` path on localhost port `8081`:
 
 . In Anypoint Studio, create a new Mule project.
-. Click the Exchange icon *(X)* in the upper-left of the Studio task bar.
-. In Exchange, click *Login* and supply your Anypoint Platform username and password.
-. In Exchange, select *Connectors* as the type, and search for "SAP Connector".
-. Select the SAP Connector for Mule 4 tile, and click *Add to project*.
-. Drag the SAP *Function source* operation to the source area of the canvas at the start of a flow.
-. Create a new *SAP Inbound* global element configuration and specify the environment values.
-. Add a *Transform Message* component to the flow and specify the details based on the metadata:
-+
-image::sap-function-return-response.png[]
-+
-The flow now contains *Function source* in the canvas source area and the *Transform Message*, as shown in the following image:
-+
-image::sap-flow-response.png[]
+. In the *Mule Palette* view, drag the *HTTP > Listener* operation to the canvas to start the flow.
+. Configure the global element using the default values.
+. In the *Listener* properties tab, set the path to `/sendIDoc`.
+
+=== Add the Transform Message Component
+
+Configure the *Transform Message* component to convert the IDoc input to XML:
+
+. From the *Mule Palette* view, select *Core* and drag the *Transform Message* operation to the right of the *Listener* component.
+. After the metadata is retrieved, overlay the text in *Output* section of the *Transform Message* Component with the following text:
 +
-The SAP Design Studio displays feedback about the received object and its response:
+*Reviewers, is this shortened text OK?*
```

Comment:
Please use the full idoc
<MATMAS01>
    <IDOC BEGIN="1">
        <EDI_DC40 SEGMENT="1">
            <TABNAM>EDI_DC40</TABNAM>
            <MANDT>800</MANDT>
            <DOCNUM>0000000003519646</DOCNUM>
            <DOCREL>740</DOCREL>
            <STATUS>30</STATUS>
            <DIRECT>1</DIRECT>
            <OUTMOD>2</OUTMOD>
            <IDOCTYP>MATMAS01</IDOCTYP>
            <MESTYP>MATMAS</MESTYP>
            <SNDPOR>SAPIDE</SNDPOR>
            <SNDPRT>LS</SNDPRT>
            <SNDPRN>T90CLNT090</SNDPRN>
            <RCVPOR>MULE01_TP</RCVPOR>
            <RCVPRT>LS</RCVPRT>
            <RCVPRN>MULE01_LS</RCVPRN>
            <CREDAT>20180606</CREDAT>
            <CRETIM>133420</CRETIM>
            <SERIAL>20180606133420</SERIAL>
        </EDI_DC40>
        <E1MARAM SEGMENT="1">
            <MSGFN>005</MSGFN>
            <MATNR>000000000000000088</MATNR>
            <ERSDA>19970527</ERSDA>
            <ERNAM>MORLEY</ERNAM>
            <LAEDA>20030122</LAEDA>
            <AENAM>I021066</AENAM>
            <PSTAT>KVB</PSTAT>
            <LVORM>X</LVORM>
            <MTART>FERT</MTART>
            <MBRSH>M</MBRSH>
            <MATKL>02004</MATKL>
            <MEINS>PCE</MEINS>
            <BLANZ>000</BLANZ>
            <BRGEW>0.200</BRGEW>
            <NTGEW>0.200</NTGEW>
            <GEWEI>KGM</GEWEI>
            <VOLUM>0.000</VOLUM>
            <TRAGR>0001</TRAGR>
            <SPART>09</SPART>
            <WESCH>0.000</WESCH>
            <LAENG>0.000</LAENG>
            <BREIT>0.000</BREIT>
            <HOEHE>0.000</HOEHE>
            <ERGEW>0.000</ERGEW>
            <ERVOL>0.000</ERVOL>
            <GEWTO>0.0</GEWTO>
            <VOLTO>0.0</VOLTO>
            <FUELG>0</FUELG>
            <STFAK>0</STFAK>
            <MHDRZ>0</MHDRZ>
            <MHDHB>0</MHDHB>
            <MHDLP>0</MHDLP>
            <VPSTA>KVB</VPSTA>
            <MSTDE>00000000</MSTDE>
            <MSTDV>00000000</MSTDV>
            <COMPL>00</COMPL>
            <GEWTO_NEW>0.0</GEWTO_NEW>
            <VOLTO_NEW>0.0</VOLTO_NEW>
            <ANP>000000000</ANP>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>1</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>ZH</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>4</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>RO</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>5</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>SL</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>6</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>HR</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>E</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>EN</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>F</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>FR</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>G</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>EL</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>J</SPRAS>
                <MAKTX>AS-100 T ???</MAKTX>
                <SPRAS_ISO>JA</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>W</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>BG</SPRAS_ISO>
            </E1MAKTM>
            <E1MAKTM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <SPRAS>d</SPRAS>
                <MAKTX>AS-100 T-shirt</MAKTX>
                <SPRAS_ISO>SH</SPRAS_ISO>
            </E1MAKTM>
            <E1MARCM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <WERKS>1000</WERKS>
                <PSTAT>V</PSTAT>
                <LVORM>X</LVORM>
                <PLIFZ>0</PLIFZ>
                <WEBAZ>0</WEBAZ>
                <PERKZ>M</PERKZ>
                <AUSSS>0.00</AUSSS>
                <MINBE>0.000</MINBE>
                <EISBE>0.000</EISBE>
                <BSTMI>0.000</BSTMI>
                <BSTMA>0.000</BSTMA>
                <BSTFE>0.000</BSTFE>
                <BSTRF>0.000</BSTRF>
                <MABST>0.000</MABST>
                <LOSFX>0</LOSFX>
                <AUSDT>00000000</AUSDT>
                <BEARZ>0.00</BEARZ>
                <RUEZT>0.00</RUEZT>
                <TRANZ>0.00</TRANZ>
                <BASMG>0.000</BASMG>
                <DZEIT>0</DZEIT>
                <MAXLZ>0</MAXLZ>
                <UEETO>0.0</UEETO>
                <UNETO>0.0</UNETO>
                <WZEIT>0</WZEIT>
                <VZUSL>0.00</VZUSL>
                <UMLMC>0.000</UMLMC>
                <LGRAD>0.0</LGRAD>
                <OBJID>00000000</OBJID>
                <MTVFP>01</MTVFP>
                <VRVEZ>0.00</VRVEZ>
                <VBAMG>0.000</VBAMG>
                <VBEAZ>0.00</VBEAZ>
                <TRAME>0.000</TRAME>
                <FXHOR>000</FXHOR>
                <VINT1>000</VINT1>
                <VINT2>000</VINT2>
                <LOSGR>0.000</LOSGR>
                <KAUSF>0.00</KAUSF>
                <TAKZT>0</TAKZT>
                <VRBDT>00000000</VRBDT>
                <VRBFK>0.00</VRBFK>
                <PREND>00000000</PREND>
                <PRENG>00000000</PRENG>
                <PRFRQ>0</PRFRQ>
                <SHZET>00</SHZET>
                <MMSTD>00000000</MMSTD>
                <DPLHO>0</DPLHO>
                <MINLS>0.000</MINLS>
                <MAXLS>0.000</MAXLS>
                <FIXLS>0.000</FIXLS>
                <LTINC>0.000</LTINC>
                <COMPL>00</COMPL>
                <EISLO>0.000</EISLO>
            </E1MARCM>
            <E1MARCM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <WERKS>2300</WERKS>
                <PSTAT>VB</PSTAT>
                <LVORM>X</LVORM>
                <PLIFZ>0</PLIFZ>
                <WEBAZ>0</WEBAZ>
                <PERKZ>M</PERKZ>
                <AUSSS>0.00</AUSSS>
                <MINBE>0.000</MINBE>
                <EISBE>0.000</EISBE>
                <BSTMI>0.000</BSTMI>
                <BSTMA>0.000</BSTMA>
                <BSTFE>0.000</BSTFE>
                <BSTRF>0.000</BSTRF>
                <MABST>0.000</MABST>
                <LOSFX>0</LOSFX>
                <AUSDT>00000000</AUSDT>
                <BEARZ>0.00</BEARZ>
                <RUEZT>0.00</RUEZT>
                <TRANZ>0.00</TRANZ>
                <BASMG>0.000</BASMG>
                <DZEIT>0</DZEIT>
                <MAXLZ>0</MAXLZ>
                <UEETO>0.0</UEETO>
                <UNETO>0.0</UNETO>
                <WZEIT>0</WZEIT>
                <VZUSL>0.00</VZUSL>
                <UMLMC>0.000</UMLMC>
                <LADGR>0001</LADGR>
                <LGRAD>0.0</LGRAD>
                <OBJID>00000000</OBJID>
                <MTVFP>01</MTVFP>
                <VRVEZ>0.00</VRVEZ>
                <VBAMG>0.000</VBAMG>
                <VBEAZ>0.00</VBEAZ>
                <TRAME>0.000</TRAME>
                <FXHOR>000</FXHOR>
                <VINT1>000</VINT1>
                <VINT2>000</VINT2>
                <LOSGR>0.000</LOSGR>
                <KAUSF>0.00</KAUSF>
                <TAKZT>0</TAKZT>
                <VRBDT>00000000</VRBDT>
                <VRBFK>0.00</VRBFK>
                <PRENO>00000000</PRENO>
                <PREND>00000000</PREND>
                <PRENG>00000000</PRENG>
                <PRFRQ>0</PRFRQ>
                <SHZET>00</SHZET>
                <MMSTD>00000000</MMSTD>
                <DPLHO>0</DPLHO>
                <MINLS>0.000</MINLS>
                <MAXLS>0.000</MAXLS>
                <FIXLS>0.000</FIXLS>
                <LTINC>0.000</LTINC>
                <COMPL>00</COMPL>
                <EISLO>0.000</EISLO>
            </E1MARCM>
            <E1MARMM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <MEINH>PCE</MEINH>
                <UMREZ>1</UMREZ>
                <UMREN>1</UMREN>
                <LAENG>0.000</LAENG>
                <BREIT>0.000</BREIT>
                <HOEHE>0.000</HOEHE>
                <VOLUM>0.000</VOLUM>
                <BRGEW>0.200</BRGEW>
                <GEWEI>KGM</GEWEI>
                <NEST_FTR>0</NEST_FTR>
                <MAX_STACK>0</MAX_STACK>
                <CAPAUSE>0.000</CAPAUSE>
            </E1MARMM>
            <E1MBEWM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <BWKEY>2300</BWKEY>
                <LVORM>X</LVORM>
                <VPRSV>S</VPRSV>
                <VERPR>0</VERPR>
                <STPRS>0.3</STPRS>
                <PEINH>1</PEINH>
                <BKLAS>7920</BKLAS>
                <VMVPR>S</VMVPR>
                <VMVER>0</VMVER>
                <VMSTP>0.3</VMSTP>
                <VMPEI>1</VMPEI>
                <VMBKL>7920</VMBKL>
                <VJVPR>S</VJVPR>
                <VJVER>0</VJVER>
                <VJSTP>0.3</VJSTP>
                <LFGJA>1998</LFGJA>
                <LFMON>09</LFMON>
                <ZKPRS>0</ZKPRS>
                <ZKDAT>00000000</ZKDAT>
                <BWPRS>0</BWPRS>
                <BWPRH>0</BWPRH>
                <VJBWS>0</VJBWS>
                <VJBWH>0</VJBWH>
                <VVJLB>0.000</VVJLB>
                <VVMLB>0.000</VVMLB>
                <VVSAL>0</VVSAL>
                <ZPLPR>0</ZPLPR>
                <ZPLP1>0</ZPLP1>
                <ZPLP2>0</ZPLP2>
                <ZPLP3>0</ZPLP3>
                <ZPLD1>00000000</ZPLD1>
                <ZPLD2>00000000</ZPLD2>
                <ZPLD3>00000000</ZPLD3>
                <BWPH1>0</BWPH1>
                <BWPS1>0</BWPS1>
                <ABWKZ>00</ABWKZ>
                <PSTAT>B</PSTAT>
                <KALN1>000100014878</KALN1>
                <KALNR>000100014879</KALNR>
                <VERS1>00</VERS1>
                <VERS2>00</VERS2>
                <VERS3>00</VERS3>
                <PPRDZ>000</PPRDZ>
                <PPRDL>000</PPRDL>
                <PPRDV>000</PPRDV>
                <PDATZ>0000</PDATZ>
                <PDATL>0000</PDATL>
                <PDATV>0000</PDATV>
                <VPLPR>0</VPLPR>
                <VJBKL>7920</VJBKL>
                <VJPEI>1</VJPEI>
                <HKMAT>X</HKMAT>
                <BWPEI>0</BWPEI>
            </E1MBEWM>
            <E1MLANM SEGMENT="1">
                <MSGFN>005</MSGFN>
                <ALAND>ES</ALAND>
                <TATY1>MWST</TATY1>
                <TAXM1>0</TAXM1>
            </E1MLANM>
        </E1MARAM>
    </IDOC>
</MATMAS01>